dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -263,52 +276,56 @@ public Response canLockContent(@Context HttpServletRequest request,
             }
         }
 
-        Contentlet contentlet = (inode != null)
-                ? APILocator.getContentletAPI().find(inode, user, live)
-                : APILocator.getContentletAPI()
-                        .findContentletByIdentifier(id, live, lang, user, live);
-        if (contentlet == null || contentlet.getIdentifier() == null) {
-            jo.append("message", "contentlet not found");
-            jo.append("return", 404);
+        try {
+            Contentlet contentlet = (inode != null)
+                    ? APILocator.getContentletAPI().find(inode, user, live)
+                    : APILocator.getContentletAPI()
+                            .findContentletByIdentifier(id, live, lang, user, live);
+            if (contentlet == null || contentlet.getIdentifier() == null) {
+                jo.append("message", "contentlet not found");
+                jo.append("return", 404);
+
+                Response.ResponseBuilder responseBuilder = Response.status(HttpStatus.SC_NOT_FOUND);
+                return responseBuilder.entity(jo).build();
+            } else {
+                if (!UtilMethods.isSet(inode)) {
+                    inode = contentlet.getInode();
+                }
+                if (!UtilMethods.isSet(id)) {
+                    id = contentlet.getIdentifier();
+                }
 
-            Response.ResponseBuilder responseBuilder = Response.status(HttpStatus.SC_NOT_FOUND);
-            return responseBuilder.entity(jo).build();
-        } else {
-            if (!UtilMethods.isSet(inode)) {
-                inode = contentlet.getInode();
-            }
-            if (!UtilMethods.isSet(id)) {
-                id = contentlet.getIdentifier();
-            }
+                boolean canLock = false;
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)